### ISSUE TITLE
XSS Javascript Atack on jtable

### ISSUE THREAD

**Issue by tmandys** [hikalkan/jtable#1651]
When testing protection against XSS Javascript execution attacks I added text 
`<script type="text/javascript"> alert('alert'); </script>` 
as text into a field (via normal edit dialog) which is also displayed in list. Unfortunately the Javascript is executed. Surprisingly it's not executed in dialog box (`input` or `textarea`). I believe the protection using escaping should be implemented in `_getDisplayTextForRecordField()` function. But it's not trivial because `display` function may legally return also HTML. Can anybody who is familiar with side effects look at the security hole ?

**Comment by sheryever** [comment]
@tmandys 

XSS Javascript should be handle at server side and if server is returning javascript then this is server side then it is fine. it should be considered as @Html.Raw() data :)

jtable create and update forms first submit the data to server then update the grid from the `Record`  object of create/update action result `{Result: "OK", Record: { /* new or updated record object */ }}`

I recommend to everyone to always return `Record` object in your create/update result.

**Comment by tmandys** [comment]
I disagree, for XSS protection is responsible who is rendering HTML, see Smarty's `escape` function. In jtable you can even make self XSS attack. Just edit any record put `javascript` code in a field which is listed and post record. I can see `update` Ajax is sent, listing is updated using internal edit form data and javascript is executed (probably not problem of insert because new record is expected in JSON response). There is no chance to filter anything on server if user can legally enter HTML tags. The only solution in current implementation could be error response "Error: escape manually all "<" charracters". In jtable examples it does not occur because update returns error when appears < followed by slash, ! or letter,  but it's buggy - see what illegal HTML text is responded from server.

**Comment by jolentes** [comment]
We do HTML encode data on the server before display but when using the edit form its fields are filled with the encoded strings. We ran into that issue with umlauts, not even any html special chars.
